bot.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"github.com/domino14/macondo/bot"
+	"github.com/domino14/macondo/config"
+	"github.com/domino14/macondo/runner"
+)
+
+const (
+	GracefulShutdownTimeout = 20 * time.Second
+)
+
+var profilePath = flag.String("profilepath", "", "path for profile")
+
+func main() {
```

Comment:
If `bot.go` is at the top level I don't think this will work, because main.go already defines a `main` function. Did this compile?

Reply:
it compiles since it's a separate entry point and does not link to main.go (mostly i didn't follow everything that was going on in main.go in terms of what was needed and what wasn't, so i just copied it into bot.go and then changed the bit that started the shell to start a bot instead). i'll move it.

Reply:
moved it